examples/quick_start/scripts/preproc/preprocessing.py

```diff
@@ -370,51 +440,56 @@ def run(self):
         # targets might not be available for test/predict_dataset
         train_dataset_features = nvt_workflow_features.fit_transform(train_dataset)
         train_dataset_targets = nvt_workflow_targets.fit_transform(train_dataset)
-        train_dataset_preproc = nvt.Dataset(
-            self.df_lib.concat(
-                [train_dataset_features.to_ddf(), train_dataset_targets.to_ddf()],
-                axis=1,
-            ),
-            schema=train_dataset_features.schema + train_dataset_targets.schema,
-            cpu=not self.gpu,
-        )
 
         output_train_dataset_path = os.path.join(output_dataset_path, "train")
-        logging.info(f"Fitting and transforming train set: {output_train_dataset_path}")
+        logging.info(f"Saving transformed train set: {output_train_dataset_path}")
+        train_dataset_preproc = self.merge_dataset_features_values(
+            train_dataset_features, train_dataset_targets, "train", args
+        )
         train_dataset_preproc.to_parquet(
             output_train_dataset_path,
             output_files=args.output_num_partitions,
         )
 
         if args.eval_data_path or args.dataset_split_strategy:
+            logging.info("Transforming the eval set")
             eval_dataset = nvt.Dataset(eval_ddf, cpu=not self.gpu)
             # Processing features and targets in separate workflows, because
             # targets might not be available for test/predict_dataset
             eval_dataset_features = nvt_workflow_features.transform(eval_dataset)
             eval_dataset_targets = nvt_workflow_targets.transform(eval_dataset)
-            eval_dataset_preproc = nvt.Dataset(
-                self.df_lib.concat(
-                    [eval_dataset_features.to_ddf(), eval_dataset_targets.to_ddf()],
-                    axis=1,
-                ),
-                schema=eval_dataset_features.schema + eval_dataset_targets.schema,
-                cpu=not self.gpu,
-            )
 
             output_eval_dataset_path = os.path.join(output_dataset_path, "eval")
-            logging.info(f"Transforming eval set: {output_eval_dataset_path}")
-
+            logging.info(f"Saving transformed eval set: {output_eval_dataset_path}")
+            eval_dataset_preproc = self.merge_dataset_features_values(
+                eval_dataset_features, eval_dataset_targets, "eval", args
+            )
             eval_dataset_preproc.to_parquet(
                 output_eval_dataset_path,
                 output_files=args.output_num_partitions,
             )
 
         if args.predict_data_path:
+            # Adding to predict set dummy target columns that are
```

Comment:
does not read well.  may be rephrase as `Adding a dummy target column(s) to the test set to perform target encoding op while this issue ...`